mypyc/ops.py

```diff
@@ -1576,8 +1579,17 @@ def get_method_and_class(self, name: str) -> Optional[Tuple[FuncIR, 'ClassIR']]:
         for ir in self.mro:
             if name in ir.methods:
                 return ir.methods[name], ir
+
             if name in ir.properties:
-                return ir.properties[name], ir
+                return ir.properties[name][0], ir
+
+            if name.startswith(PROPSET_PREFIX):
+                prop_name = name.split(PROPSET_PREFIX)[1]
+                if prop_name in ir.properties:
+                    setter = ir.properties[prop_name][1]
+                    if setter:
+                        return (setter, ir)
```

Comment:
I don't love this whole special casing here, though probably a bunch of the problem is the distinction between methods and properties is always kind of blurry anyways.

Maybe the thing to do is to always store properties in `methods` as well, which gets rid of the need to search `properties` here and when creating vtables.

Probably then we'll also need to track whether a function is a property in its `FuncDecl` to avoid emitting properties as methods, but that seems reasonable.